app/src/main/java/com/xmartlabs/template/usecase/BaseSingleUseCase.kt

```diff
@@ -0,0 +1,65 @@
+package com.xmartlabs.template.usecase
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.xmartlabs.template.usecase.internal.DefaultScheduler
+import com.xmartlabs.template.usecase.internal.Scheduler
+import io.reactivex.Single
+import timber.log.Timber
+
+abstract class BaseSingleUseCase<in P, R> {
+  protected var taskScheduler: Scheduler = DefaultScheduler
+
+  /** Executes the use case asynchronously and places the [Result] in a MutableLiveData
+   *
+   * @param parameters the input parameters to run the use case with
+   * @param result the MutableLiveData where the result is posted to
+   *
+   */
+  operator fun invoke(parameters: P, result: MutableLiveData<Result<R>>) {
+    // result.value = Result.Loading TODO: add data to Loading to avoid glitches
+    try {
+      taskScheduler.execute {
+        try {
+          execute(parameters).let { useCaseSingleResult ->
+            result.postValue(Result.success(useCaseSingleResult.blockingGet())) // TODO: Fix it in the future
+          }
+        } catch (e: Exception) {
```

Comment:
is Exception needed in this case? would be better if we use a more specific exception

Reply:
it's the base usecase, so I think you cannot use an specific exception right?